### ISSUE TITLE
Webpack plugin: Only create precache manifest for a specified entrypoint

### ISSUE THREAD

**Issue by qqilihq** [GoogleChrome/workbox#2039]
**Library Affected**:
workbox-webpack-plugin

**Issue or Feature Request Description**:

We’ve a Webpack configuration with several entry points which represent (somewhat) isolated parts. I’d like to build the precache manifest JSON file only for the files which are being used in **one** entry point of the project -- files and assets (managed via `file-loader`) from other entry points should **not** appear in the generated manifest.

I tried using both, the `chunks` and the `excludeChunks` settings, but neither give me the desired result:

* when I use `chunks` and set it to the relevant entry point (e.g. `entrypoint1`), my manifest does not contain the assets (e.g. images) which are referenced in the entry point

* when I use `excludeChunks`, the manifest contains **all** assets (e.g. all images from the entire project, no matter whether in which entrypoint they're being referenced)

A minimal example project which demonstrates my issue can be found here:
https://github.com/qqilihq/workbox-chunk-issue

**Steps to reproduce:**

1. Clone the repo
2. `yarn`
3. `yarn build`
4. Check `dist/precache-manifest.af8924052142092476d2b7a9665c1d2e.js` -- I'd expect it to contain an image beside the JS file
5. Modify `webpack.config.js` and toggle comments in line 18 and 19 to use `excludeChunks`
6. Check `dist/precache-manifest.89bef461327312c73acadd63bd01a11a.js` -- I'd epxect it to contain **one** image (the one from `entrypoint1.js`), however it also contains an image from `entrypoint2.js`

**Comment by jeffposnick** [comment]
This sounds similar to #1859. I'll take this into account when testing out the new Workbox v5 behavior.

**Comment by jeffposnick** [comment]
Looking into this a little more, when I try a webpack build, the output is:

```sh
Hash: 24f9195c1c0424db6903
Version: webpack 4.35.2
Time: 794ms
Built at: 07/02/2019 2:04:15 PM
                                      Asset      Size       Chunks             Chunk Names
        entrypoint1.6e7b84eea984685841ae.js  4.35 KiB  entrypoint1  [emitted]  entrypoint1
        entrypoint2.79bbf4af966bc2af7d10.js  4.34 KiB  entrypoint2  [emitted]  entrypoint2
kitten.97c7ec9e215a8d0d51f1e7205c94c50c.jpg   904 KiB               [emitted]  
 puppy.3fc56dd1d4dd7ef3a30cb9a56b537fd5.jpg  34.6 KiB               [emitted]  
                          service-worker.js  2.44 KiB               [emitted]  
                        workbox-1997d11f.js  55.7 KiB               [emitted]  
Entrypoint entrypoint1 = entrypoint1.6e7b84eea984685841ae.js
Entrypoint entrypoint2 = entrypoint2.79bbf4af966bc2af7d10.js
[./src/entrypoint1.js] 95 bytes {entrypoint1} [built]
[./src/entrypoint2.js] 92 bytes {entrypoint2} [built]
[./src/images/kitten.jpg] 89 bytes {entrypoint1} [built]
[./src/images/puppy.jpg] 88 bytes {entrypoint2} [built]
```

You'll notice that neither of the `.jpg` files have a value in the `Chunk Names` column.

Similarly, when I look into the compilation stats inside of the v5 webpack plugin, I can't find anything that maps those `.jpg` files to a named chunk:

<img width="505" alt="Screen Shot 2019-07-02 at 2 27 48 PM" src="https://user-images.githubusercontent.com/1749548/60537086-b3cc7980-9cd5-11e9-995a-1dd6b4f355a5.png">

(Apologies for sharing a screenshot, but that's the best representation I could find.)

My conclusion is that the `file-loader` doesn't actually create a chunk association when it adds the file assets to the webpack compilation. There's unfortunately no way for the Workbox plugins to infer that association.

https://github.com/webpack-contrib/file-loader/issues/28 seems to imply that that's the case as well.

I think you're going to have to use `exclude`/`include` instead of something `chunk`-based.

**Comment by jeffposnick** [comment]
This should be addressed by the current Workbox v5.0.0 alpha.
